### PR TITLE
Basic binary tree with traversals. 

### PR DESCRIPTION
**Pull request by cyberwizard1001**
Basic binary tree with traversal algorithms working. Hardcoded inputs for now, will be adding more functions and custom input in the following days.

### CODE REVIEW

**Review by aayushjain7 — DISMISSED** [review]

**Review by aayushjain7 — COMMENTED** [review]
resolve conflicts.

File: trees/python/binary-tree.py [hunk 1]
```diff
@@ -0,0 +1,66 @@
+from collections import deque
+
+class Node(object):
```
Comment (Arsenic-ATG) [inline trees/python/binary-tree.py:3]: According to [PEP8](https://www.python.org/dev/peps/pep-0008/), there should be a 2 line separation between your import statements and rest of your code.

Comment (Arsenic-ATG) [inline trees/python/binary-tree.py:3]: Following [PEP8](https://www.python.org/dev/peps/pep-0008/), there should be a 2 line separation between your import statement and rest of your code.

**Review by Arsenic-ATG — CHANGES_REQUESTED** [review]

File: trees/python/binary-tree.py [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+    def __init__(self,root):
+        self.root = Node(root)
+
+    def PreOrderTraversal(self,v):
+        if v!=None:
+            print(v.value,end=" ")
+            self.PreOrderTraversal(self,v.leftchild)
+            self.PreOrderTraversal(self,v.rightchild)
+
+        return
```
Comment (cclauss) [inline trees/python/binary-tree.py:19]:
```suggestion
    def PreOrderTraversal(self, node: Node) -> None:
        if node:
            print(node.value, end=" ")
            self.PreOrderTraversal(node.leftchild)
            self.PreOrderTraversal(node.rightchild)
```
This function accepts a `Node`, not a value so `v` is not a great parameter name.  Besides, single-letter variable names make code look old-fashoned so where possible please use variable names that make code comments unnecessary.

File: trees/python/binary-tree.py [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+        	if temp.leftchild!=None:
+        		q1.append(temp.leftchild)
+        	if temp.rightchild!=None:
+        		q1.append(temp.rightchild)
+        return
+
+    def PrintTree(self,nlist):
+        for i in range(len(nlist)):
+            if (nlist[i] != None):
+                print(nlist[i].element,end=" ")
```
Comment (cclauss) [inline trees/python/binary-tree.py:52]:
```suggestion
        for item in nlist:
            if item:
                print(item.element, end=" ")
```

File: trees/python/binary-tree.py [hunk 4]
```diff
@@ -0,0 +1,66 @@
+from collections import deque
```
Comment (cclauss) [inline trees/python/binary-tree.py:1]: Please rename the file to `algorithms/Python/trees/binary_tree.py` to march the changes made in #207

File: trees/python/binary-tree.py (hunk 1 above)
Comment (cclauss) [reply trees/python/binary-tree.py:3]: To automate this process try:
```
python3 -m pip install black isort  # this line only needs to be done once
black my_file.py
isort --profile black my_file.py
```

File: trees/python/binary-tree.py [hunk 5]
```diff
@@ -0,0 +1,66 @@
...
+                print(-1)
+
+tree = BinaryTree(1)
+tree.root.left = Node(2)
+tree.root.right = Node(3)
+tree.root.left.left = Node(4)
+tree.root.left.right = Node(5)
+tree.root.right.left = Node(6)
+tree.root.right.right = Node(7)
+
```
Comment (cclauss) [inline trees/python/binary-tree.py:63]: Please add calls to `PreOrderTraversal()`, `InOrderTraversal()`, `PostOrderTraversal()`, `levelorderTraverse()`, and  `PrintTree()` so we are sure that they all work as expected.  Python methods names should be snake_case so `pre_order_traversal()`, not `PreOrderTraversal()`

File: trees/python/binary-tree.py (hunk 1 above)
Comment (cyberwizard1001) [reply trees/python/binary-tree.py:4]: Ah didn't note that. Thank you! Modified.

File: trees/python/binary-tree.py (hunk 3 above)
Comment (cyberwizard1001) [reply trees/python/binary-tree.py]: I have removed the print function for now. Will keep these in mind while reintroducing the function.

File: trees/python/binary-tree.py (hunk 5 above)
Comment (cyberwizard1001) [reply trees/python/binary-tree.py]: Done!

**Comment by cclauss** [comment]
% `python3 trees/python/binary_tree.py`
```
Traceback (most recent call last):
  File "/Users/cclauss/Python/binary_tree.py", line 65, in <module>
    tree.pre_order_traversal(tree.root)
  File "/Users/cclauss/Python/binary_tree.py", line 17, in pre_order_traversal
    self.PreOrderTraversal(self,nnode.leftchild)
AttributeError: 'BinaryTree' object has no attribute 'PreOrderTraversal'
```

**Review by sridhar-5 — APPROVED** [review]
The code looks good to me.

**Comment by cyberwizard1001** [comment]
Sorry bout that, fixed now.

**Comment by cclauss** [comment]
Let’s not be in rush to merge a PR that has failing tests.

When I run the program, I expect to see 1, 2, 3 or 1, 3, 2 or 2, 1, 3, or 2, 3, 1 or 3, 1, 2 or 3, 2, 1 for each traversal.  I just see 1s so the results are unclear to me.

**Comment by cclauss** [comment]
Let's make sure that the output matches https://www.geeksforgeeks.org/tree-traversals-inorder-preorder-and-postorder or some other trusted source.

**Review by cclauss — CHANGES_REQUESTED** [review]
Let's make sure that the program's output matches https://www.geeksforgeeks.org/tree-traversals-inorder-preorder-and-postorder or some other trusted source.

**Comment by ming-tsai** [comment]
Hi @cyberwizard1001,
Sorry about this, please move the file to `algorithms/Python/tree` folder

**Comment by cyberwizard1001** [comment]
Sure. I'm not sure why this doesn't work, was working before I pushed and altered. I haven't changed the algorithm or how it works so not sure what the issue is. Will close this commit, work out the issues and reupload when I'm sure it works.

**Comment by ming-tsai** [comment]
We have had changed it by this discussion #206

**Comment by cclauss** [comment]
```python
tree = BinaryTree(1)
tree.root.left = Node(2)
tree.root.right = Node(3)
...
tree = BinaryTree(1)
tree.root.left_child = Node(2)
tree.root.right_child = Node(3)
```

**Pull request closed without merging**